src/Uno.UI/UI/Xaml/UIElement.Pointers.iOS.cs

```diff
@@ -176,6 +181,18 @@ public override void TouchesEnded(NSSet touches, UIEvent evt)
 					var id = NativePointer.Get(this, touch);
 					var args = new PointerRoutedEventArgs(id.Id, touch, evt, this);
 
+					if (!_hadNativeMove)
```

Comment:
The only thing we could check is the pointer ID, but as stated in the comment below, if we raise an additional 'move' it's really not a problem. Even on UWP, it's 100% acceptable to get 2 events at the same position (which is fairly easy with mouses).

Reply:
> The only thing we could check is the pointer ID

I don't understand why you can't compare the touch position with the position on pointer down, to determine if it actually moved.

> // But on UWP it seems that we always have a least one move

This doesn't seem to be the case - if the pointer doesn't move then there's no PointerMoved, just PointerPressed and PointerReleased. (With mouse at least, I didn't test touch.)

> if we raise an additional 'move' it's really not a problem

I would be worried that introducing a spurious `PointerMoved` might cause a bug for somebody, somewhere. But you know more about pointers than I do, so I'll defer to your judgement here.

Reply:
> 
> 
> > The only thing we could check is the pointer ID
> 
> I don't understand why you can't compare the touch position with the position on pointer down, to determine if it actually moved.

Yeah I was no clear: we can definitively compare location, however I really don't see any gain to do it, especially since here where we only receive **touches**, so location are always approximate / a bit offset compared to previous. It would almost never filter any event.

We could have interest to compare pointer ID to properly track the "had move" per pointer, but again it would most probably not filter anything as the gesture that would be filtered is extremely hard to reproduce (Pressing down and move finger 1, then "fast" flicking with a second finger and release first finger at almost the same time).

> 
> > // But on UWP it seems that we always have a least one move
> 
> This doesn't seem to be the case - if the pointer doesn't move then there's no PointerMoved, just PointerPressed and PointerReleased. (With mouse at least, I didn't test touch.)

It's the case for touch inputs ... according to my tests. I definitively agree that for mouse we can have only a "pressed" and a "release", but I was not able to get this neither for fingers nor for pen. I'll specify that this comment is relevant only for touch and pen.

> 
> > if we raise an additional 'move' it's really not a problem
> 
> I would be worried that introducing a spurious `PointerMoved` might cause a bug for somebody, somewhere. But you know more about pointers than I do, so I'll defer to your judgement here.

Well OSes are raising a lot of pointer events, especially UWP. They're often changing only one value at once, and it's 100% possible to have multiple events at the same location (like you noticed with a mouse). So it's fairly improbable that it would break someone's code to have 2 events at the same location (I'm thinking to a divide per zero exception for instance).

At the opposite, I would not be surprised that some (not so well designed) UWP apps would not listen to all events and instead heavily relying on the "move". This change would improve their support.

The only thing which is really problematic to raise additional events like this, is that they are sharing the same `FrameID` (a.k.a. timestamp), which is really problematic when you are trying to compute speed :/ . I realized that when I've implemented inertia, but I've only workarounded it for now and I didn't validated the UWP behavior neither (I would expect that some `PointerReleased` and `PointerCaptureLost` would share the same timestamp). But that's actually the case since the beginning of uno's support of pointer events, so one more "move" should not start to break an app (we are already raising some events like "entered" and "exited" which share the same frame ID of a "move").